### ISSUE TITLE
MLP not learning (Learn state attibutes instead of forcing attributes etc.)

### ISSUE THREAD

**Issue by RY4GIT** [NWC-CUAHSI-Summer-Institute/dCFE#26]
Discussion with @jmframe today

- [x] Use scheduling for learning rate, step size 1 and try gamma 0.7
  - https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.StepLR.html#torch.optim.lr_scheduler.StepLR
  - Probably learning rate 1e-03 is good 
- [x] Check to make sure initialization is correct 
- [x] Use at least 3 years of observation data for training 
- [x] Use state variables (e.g., GIUH queues, soil bucket level, or GW level) as MLP learning input, because we want to produce different responses to the same situation (forcing) given different initial conditions (state in the model) 
- [x] Change self.c from NLDAS attributes to forcing?

**Comment by RY4GIT** [comment]
Addressed tasks in NWC-CUAHSI-Summer-Institute/dCFE/pull/29

**Comment by RY4GIT** [comment]
Discussion with @jmframe and @taddyb today
- [x] Normalize by max & min state of reservoir
- [ ] Run CFE once before MLP in epoch 1
- [ ] Try max_gw_storage instead of refkdt
- [ ] Everytime step run MLP, use normalized c t-24 to t-0 & states at t-0 because Sequence-to-sequence doesn't work (currently timestep 0 has access to 3600)
- [ ] Move to LSTM

**Comment by RY4GIT** [comment]
Now I get errors about gradient overwritten

**Comment by RY4GIT** [comment]
Debug session with @jmframe today

What's done 
- Remove shuffle from the dataloader
- Add activation function in between the Linear layers 
- Increase the hidden size

Todo
- Learn against real streamflow case (hard to make static prediction with MLP)
- Create a plot
- Loop through each basin

**Comment by RY4GIT** [comment]
Addressed in 67e119fcce24f831846837a1da39aedb95b97d5a
